darks_driftwind_crypt/developmental_testing.dsc

```diff
@@ -0,0 +1,166 @@
+      - flag server region_five_name..players:<-:<player>
+      - if !<server.has_flag[region_five_name.players]>:
+        - remove <server.flag[region_five_name.fireball_entities].filter[exists]>
+        - flag server region_five_name:!
+
+# @ #10 I also need a boundary that will force teleport the player up if they land on it on the "fog" below
+    after player enters region_six_name:
+      - run fancy_painful_teleport defmap:teleport=fog_teleport_respawn
+
+# @ 11 I need this door to open when a player is close to it
+    after player enters region_seven_name:
+      - run open_gate_task defmap:door=two
+
+# @ #12 I need a system that will give the player the book that is currently on the lectern when they read it, and remove it when they exit the room*
+#13 I also need a system that will teleport the player to the start of the room if they step on an incorrect pressure plate
+  #- The solution is on the ceiling in wool
+
+    after player right clicks lectern:
+      - if <player.has_flag[region_six.has_book]>:
+        - stop
+      - flag player region_six.has_book
+      - give <context.location.inventory.list_contents.first>
+    after player exits region_six:
+      - if <player.inventory.contains_item[written_book]>:
+        - take item:written_book
+
+    # ! bad pressure plates would just be easily defined by flagging them via `/ex flag <player.cursor_on> bad_pressure_plate`
+    after player stands on *_pressure_plate:
+      - if <context.location.has_flag[bad_pressure_plate]>
+      - run fancy_painful_teleport defmap:teleport=region_six_teleport_respawn
+
+# @ #14 I need this door to open when the player has finished the puzzle
+    after player enters region_eight_name:
+      - run open_gate_task defmap:door=three
+
+# @ #15 I need a transition to the large hallway to the left of this block
+    after player enters region_nine_name:
+      - run fancy_teleport defmap:teleport=room_four_name
+
+#17 Npc named "Vel", what he says is tbd, need him to vanish when he finishes talking as if he died
+    # @ see below maybe
+    on npc death:
+      - if <npc.script.name.if_null[invalid]> == vel:
+        - determine "well, i'm pretty sure he was going to say Behr was the coolest ever but he died a very spontaneous and what seemed to be quick death."
+
+#18 Door that needs to be opened *after* the npc vanishes
+      - wait 3s
+      - run open_gate_task defmap:door=four
+
+#19 - No block - The entire hall is where the boss fight will take place
+
+# misc:
+    after reload scripts:
+      - define generic_location <location[0,0,0,<server.worlds.first.name>]>
+      - foreach region_one_name|region_two_name|yellow_buttons|red_buttons|blue_buttons|gray_buttons|region_four_name|region_five_name|region_six_name|region_seven_name|region_eight_name|region_nine_name as:area:
+        - if !<cuboid[<[area]>].exists>:
+          - note <[generic_location].to_cuboid[<[generic_location]>]> as:<[area]>
```

Comment:
this is strictly for the failsafe mentioned in the pull request description; this a lazy method of "create notables for events that utilize events with unnoted notables"